tests/phpunit/tests/widgets.php

```diff
@@ -686,6 +686,53 @@ public function test_wp_widget_get_settings() {
 		$this->assertArrayNotHasKey( 0, $never_used );
 	}
 
+	/**
+	 * @covers WP_Widget::get_settings
+	 * @ticket 54677
+	 */
+	public function test_wp_widget_initiates_widget_with_alt_option() {
+		/*
+		 * Pretend the recent posts widget is new.
+		 *
+		 * The widget contains an alt (legacy) option so both the
+		 * new and the old option need to be deleted.
+		 */
+		delete_option( 'widget_recent-posts' );
+		delete_option( 'widget_recent_entries' );
```

Comment:
If at some point a patch is written that causes `delete_option()` to fail in some circumstances, this test could fail without being clear on the reason.

Adding this will prevent confusing our future selves and ensure that the current state is appropriate for the test method to continue:

```php
delete_option( 'widget_recent-posts' );
$this->assertFalse( get_option( 'widget_recent-posts' ), 'The "widget_recent-posts" option was not deleted before testing' );
delete_option( 'widget_recent_entries' );
$this->assertFalse( get_option( 'widget_recent_entries' ), 'The "widget_recent_entries" option was not deleted before testing' );
```

This also applies to the other test method which uses `delete_option()` and `update_option()`.

Reply:
> If at some point a patch is written that causes `delete_option()` to fail in some circumstances, this test could fail without being clear on the reason.

Wouldn't it be better to add such tests to `tests/phpunit/tests/option` rather than here?

I understand the logic but worry that it would be adding options tests to the widget test suite.

Reply:
In this case, it's not testing the behaviour of `get_option()`, but rather that the setup and pre-existing values for the test are correct, to avoid incorrect test results due to a regression introduced later. I had a brief discussion with @jrfnl about this and best practice for stabilising tests.

Some excerpts:
> I would definitely do that
> it's about creating the test environment you need and failing with a descriptive error.